onnx/defs/math/defs.cc

```diff
@@ -2859,6 +2859,7 @@ bool BuildContextDependentFunctionBody(
   }
   auto input_type = ctx.getInputType(0)->tensor_type().elem_type();
   bool float_input = input_type == TensorProto_DataType_FLOAT;
+  std::string reduction_attr = ctx.getAttribute("reduction") == nullptr ? "mean" : ctx.getAttribute("reduction")->s();
```

Comment:
I see the reduction attribute is using mean by default: https://github.com/onnx/onnx/blob/04971f7d26aad7bbc71e2d92433631f985cb96c6/onnx/defs/math/defs.cc#L3143
I am curious that will it still be nullptr if reduction is not set?

Reply:
minor suggestion: could simplify right-hand-side to ```getAttribute(ctx, "reduction", "mean")```. See: https://github.com/onnx/onnx/blob/04971f7d26aad7bbc71e2d92433631f985cb96c6/onnx/defs/shape_inference.h#L92

Reply:
the ctx here refers to "FunctionBodyBuildContext" and not "InferenceContext"

I will use this in the type and shape inference function which uses InferenceContext

Reply:
Sorry, missed that

Reply:
@jcwchen : This code sources the attribute from InferenceContext... if the attribute is not explicitly present it will return nullptr